dali/operators/generic/permute_batch.cc

```diff
@@ -45,7 +45,8 @@ void PermuteBatch<CPUBackend>::RunImpl(HostWorkspace &ws) {
     int src = indices_[i];
     tp.AddWork([&, i, src](int tid) {
       output.SetMeta(i, input.GetMeta(i));
-      output[i].Copy(input[src]);
+      // TODO(klecki): SetSample
+      output.UnsafeCopySample(i, input, src);
```

Comment:
Why does it need to be "unsafe"? At least in case of PermuteBatch, we know the shape, so we can actualy presize the output and safely copy the samples.

Reply:
I just wanted to discourage people from using it left and right. Especially with copy, there should be nothing wrong with it, but the full intended functionality is not there yet, we just handle what we already mostly do with `tv[i].Copy(tv[j])`